src/compiler/moduleSpecifiers.ts

```diff
@@ -708,6 +712,29 @@ function getAllModulePaths(
 }
 
 function getAllModulePathsWorker(info: Info, importedFileName: string, host: ModuleSpecifierResolutionHost): readonly ModulePath[] {
+    const cache = host.getModuleResolutionCache?.();
+    const links = host.getSymlinkCache?.();
+    if (cache && links && host.readFile && !pathContainsNodeModules(info.importingSourceFileName)) {
+        Debug.type<ModuleResolutionHost>(host);
+        // Cache resolutions for all `dependencies` of the `package.json` context of the input file.
+        // This should populate all the relevant symlinks in the symlink cache, and most, if not all, of these resolutions
+        // should get (re)used.
+        const state = getTemporaryModuleResolutionState(cache.getPackageJsonInfoCache(), host, {});
+        const packageJson = getPackageScopeForPath(info.importingSourceFileName, state);
+        if (packageJson) {
+            const deps = packageJson.contents.packageJsonContent.dependencies;
+            if (deps && typeof deps === "object") {
+                const toResolve = getOwnKeys(deps as MapLike<unknown>);
+                for (const depName of toResolve) {
+                    const cached = cache.getOrCreateCacheForNonRelativeName(depName, /*mode*/ undefined, /*redirectedReference*/ undefined).get(info.canonicalSourceDirectory);
+                    if (cached) continue;
+                    const resolved = resolveModuleName(depName, info.importingSourceFileName, {}, host, cache);

```

Comment:
Do the options not matter? do we not need `resolutionMode`

Reply:
They do not - we just need any single resolution for the module name, if resolvable, under any options/settings (since we're just interested in discovering a symlink in its' folder).

Reply:
Isn’t this sensitive to our default `moduleResolution` option?